tests/integration/kubernetes/k8s-guest-pull-image.bats

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env bats
+# Copyright (c) 2023 Intel Corporation
+# Copyright (c) 2023 IBM Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/lib.sh"
+load "${BATS_TEST_DIRNAME}/../../common.bash"
+load "${BATS_TEST_DIRNAME}/tests_common.sh"
+load "${BATS_TEST_DIRNAME}/confidential_common.sh"
+
+setup() {
+    confidential_setup || skip "Test not supported for ${KATA_HYPERVISOR}."
+    setup_common
+    unencrypted_image_1="quay.io/prometheus/busybox:latest"
+    unencrypted_image_2="alpine:latest"
+}
+
+@test "Test we can pull an unencrypted image outside the guest with runc and then inside the guest successfully" {
+    [[ " ${SUPPORTED_NON_TEE_HYPERVISORS} " =~ " ${KATA_HYPERVISOR} " ]] && skip "Test not supported for ${KATA_HYPERVISOR}."
```

Comment:
I think we want to run these tests for non-TEE as well?, so this check in each test should be removed to avoid us seeing:
```
ok 1 Test we can pull an unencrypted image outside the guest with runc and then inside the guest successfully # skip Test not supported for qemu.
```
e.g. in https://github.com/kata-containers/kata-containers/actions/runs/8047127494/job/21977177697?pr=8484

Reply:
How about we enhance `run-k8s-tests-on-aks` in the next PR?

Reply:
I guess it comes back to the order of platform's readiness for support, but if for example TDX & SNP work with this PR, I don't have an issue with adding 'vanilla' `qemu` support in a separate PR

Reply:
IMHO this first implementation should be kept simple: if running on TEE then execute the test. The logic to decide whether TEE vs non-TEE is growing ugly, it is not just the hypervisor that matters but also if the workflow had other pieces of software installed (e.g. KBS, nydus...etc).

Reply:
Yeah, I think we want to avoid littering code with `BUILD_CC`, `AA_KBC` and other magic constants, but I understand your concerns. I wonder if we can have all the logic gated behind the scripts that has the master list of supported hypervisors and then use that to decide whether to deploy nydus & KBS?

Reply:
>I wonder if we can have all the logic gated behind the scripts that has the master list of supported hypervisors and then use that to decide whether to deploy nydus & KBS, so we only have one place to manage and update? I'm not sure if that is realistic due to some of them being triggered by workflows I guess, that might not interact with scripts first?

I appreciate the idea.